packages/api-client/src/views/Environment/Environment.vue

```diff
@@ -92,77 +227,95 @@ function synchronizeKeyRemoval(removedKey: string) {
+          collectionMutators.edit(
+            collection.uid,
+            'x-scalar-environments',
+            collection['x-scalar-environments'],
+          )
+        }
+      })
+    } else {
+      environmentMutators.edit(environmentId, 'color', color)
+    }
+    colorModal.hide()
   }
 }
 
-const removeEnvironment = (uid: string) => {
-  environmentMutators.delete(uid)
+const removeCollectionEnvironment = (envName: string) => {
+  activeWorkspaceCollections.value.forEach((collection) => {
+    const currentEnvironments = collection['x-scalar-environments'] || {}
+    if (envName in currentEnvironments) {
+      delete currentEnvironments[envName]
+      collectionMutators.edit(
+        collection.uid,
+        'x-scalar-environments',
+        currentEnvironments,
+      )
+    }
+  })
 
-  if (activeEnvironmentID.value === uid) {
+  if (activeEnvironmentID.value === envName) {
     const remainingEnvironments = Object.values(environments)
+    const remainingCollectionEnvironments =
+      activeWorkspaceCollections.value.flatMap((collection) =>
+        Object.keys(collection['x-scalar-environments'] || {}),
+      )
 
-    if (remainingEnvironments.length > 0) {
-      // Redirect to the last environment
+    if (remainingCollectionEnvironments.length > 0) {
+      const lastCollectionEnvironment =
+        remainingCollectionEnvironments[
+          remainingCollectionEnvironments.length - 1
+        ]
+      activeEnvironmentID.value = lastCollectionEnvironment
+      router.push({
+        name: 'environment',
+        params: { environment: lastCollectionEnvironment },
+      })
+    } else if (remainingEnvironments.length > 0) {
       const lastEnvironment =
         remainingEnvironments[remainingEnvironments.length - 1]
-
       activeEnvironmentID.value = lastEnvironment.uid
-
       router.push({
         name: 'environment',
         params: { environment: lastEnvironment.uid },
       })
     } else {
-      // Redirect to the default environment
       activeEnvironmentID.value = environments.default.uid
-
       router.push({ name: 'environment', params: { environment: 'default' } })
     }
   }
 }
 
-const handleOpenColorModal = (uid: string) => {
-  colorModalEnvironment.value = uid
-  selectedColor.value = environments[uid].color || ''
-  colorModal.show()
-}
-
-const submitColorChange = (color: string) => {
-  if (colorModalEnvironment.value) {
-    environmentMutators.edit(colorModalEnvironment.value, 'color', color)
-    colorModal.hide()
-  }
-}
-
-/** set active environment based on the route */
-const setActiveEnvironment = () => {
-  const routeEnvironmentId = router.currentRoute.value.params
-    .environment as string
-  if (routeEnvironmentId) {
-    activeEnvironmentID.value = routeEnvironmentId
-  } else if (routeEnvironmentId === 'default') {
-    activeEnvironmentID.value = environments.default.uid
+function setActiveEnvironment() {
```

Comment:
This sort of setting logic should live in the store. We may want to switch the environment from multiple places.